src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -63,20 +67,30 @@ public DashboardWebApplication(Action<WebApplicationBuilder>? configureBuilder =
         builder.Logging.AddFilter("Microsoft.AspNetCore.Server.Kestrel", LogLevel.Error);
 #endif
 
-        var dashboardConfig = LoadDashboardConfig(builder.Configuration);
+        if (builder.Configuration[DashboardConfigNames.DashboardConfigFilePathName.ConfigKey] is { Length: > 0 } configFilePath)
+        {
+            builder.Configuration.AddJsonFile(configFilePath, optional: true, reloadOnChange: true);
+        }
+
+        var dashboardConfigSection = builder.Configuration.GetSection("Dashboard");
+        builder.Services.Configure<DashboardOptions>(dashboardConfigSection);
+        builder.Services.AddSingleton<IPostConfigureOptions<DashboardOptions>, PostConfigureDashboardOptions>();
+        builder.Services.AddSingleton<IValidateOptions<DashboardOptions>, ValidateDashboardOptions>();
```

Comment:
This is fine, but there are methods to make this easier. 
```
builder.Services.AddOptions<T>().PostConfigure(..).ValidateOptions(...)
```

Reply:
I looked at them but they're based around registering funcs. I wanted to register types.